### ISSUE TITLE
DOC: ceil, floor, etc. changes in 2.0 not represented in docs

### ISSUE THREAD

**Issue by AnonymousPlayer2000** [numpy/numpy#27969]
### Describe the issue:

As [ceil's doc](https://numpy.org/doc/2.1/reference/generated/numpy.ceil.html) claims:
> The return is the ceiling of each element in x, with float dtype. This is a scalar if x is a scalar.

However, I found that NumPy-2 breaks this contract.

### Reproduce the code example:

```python
import numpy as np
a = np.array([1], dtype="int64")
b = np.ceil(a)
print(b.dtype)
```

### Error message:

```shell
>>> import numpy as np
>>> a = np.array([1], dtype="int64")
>>> b = np.ceil(a)
>>> print(b.dtype)
int64


I test it in NumPy 1.20.3:
>>> import numpy as np
>>> a = np.array([1], dtype="int64")
>>> b = np.ceil(a)
>>> print(b.dtype)
float64
```

### Python and NumPy Versions:

2.1.0
3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]

### Runtime Environment:

[{'numpy_version': '2.1.0',
  'python': '3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]',
  'uname': uname_result(system='Linux', node='42e7fc5f4d8e', release='6.8.0-49-generic', version='#49~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Nov  6 17:42:15 UTC 2', machine='x86_64')},
 {'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_KNL',
                                    'AVX512_KNM',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}},
 {'architecture': 'Haswell',
  'filepath': '/usr/local/lib/python3.10/dist-packages/numpy.libs/libscipy_openblas64_-ff651d7f.so',
  'internal_api': 'openblas',
  'num_threads': 24,
  'prefix': 'libscipy_openblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.27'}]

**Comment by seberg** [comment]
I appreciate opening the recent issues.  When you run into these things please keep in mind that there were slightly bigger changes in 2.0 and that docs can be outdated (which doesn't mean they shouldn't be fixed of course).

Doc improvements are of course always welcome, some of these functions could have had a `.. versionchanged::`.

I'll close the second issue, there is no reason to open an issue for individual functions here when it applies to all of those changed.

**Comment by Impaler343** [comment]
@seberg Should this issue be closed? As I think the related problems have been fixed

**Comment by seberg** [comment]
Yeah, dunno when it was fixed.  FWIW, the docs should have a `.. versionchanged::` there, but it is pretty niche.

It seems the actual docs may only have been updated with 2.2.
